test/JDBC/expected/todatetimeoffset.out

```diff
@@ -0,0 +1,567 @@
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 16:2a:51.766890',340)
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 16:23:5d.766890',841)
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 16:23:51.7668c0',-342)
+GO
+~~START~~
+datetimeoffset
+2000-04-22 16:23:51.7668000 -05:42
+~~END~~
+
+
+Select todatetimeoffset('2000-05-22 16:23:51.766890',234)
+GO
+~~START~~
+datetimeoffset
+2000-05-22 16:23:51.7668900 +03:54
+~~END~~
+
+
+Select todatetimeoffset('2000-04-22 16:23:51.766890',345)
+GO
+~~START~~
+datetimeoffset
+2000-04-22 16:23:51.7668900 +05:45
+~~END~~
+
+
+Select todatetimeoffset('1-04-22 16:23:51.766890',-4556)
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: The timezone provided to builtin function todatetimeoffset is invalid.)~~
+
+
+SELECT TODATETIMEOFFSET(cast('2023-08-08 16:06:45.3682170' as datetime2), -345)
+GO
+~~START~~
+datetimeoffset
+2023-08-08 16:06:45.3682170 -05:45
+~~END~~
+
+
+SELECT TODATETIMEOFFSET(cast('2023-08-08 16:06:45' as datetime2), -234)
+GO
+~~START~~
+datetimeoffset
+2023-08-08 16:06:45.0000000 -03:54
+~~END~~
+
+
+SELECT TODATETIMEOFFSET(cast('2023-08-08' as datetime2), 4556)
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: The timezone provided to builtin function todatetimeoffset is invalid.)~~
+
+
+Select todatetimeoffset('0001-01-00 00:00:00.00', '-10:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('0001-01-01 00:00:00.00', '+13:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('9999-12-31 23:59:59.59', '+12:00')
+GO
+~~START~~
+datetimeoffset
+9999-12-31 23:59:59.5900000 +12:00
+~~END~~
+
+
+Select todatetimeoffset('9999-12-31 23:59:59.59', '+12:00')
+GO
+~~START~~
+datetimeoffset
+9999-12-31 23:59:59.5900000 +12:00
+~~END~~
+
+
+Select todatetimeoffset('10000-12-31 23:59:59.59', '+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('9934-11-30 22:52:59.59', '+14:01')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: The timezone provided to builtin function todatetimeoffset is invalid.)~~
+
+
+Select todatetimeoffset('9934-11-30 22:52:59.59', '-14:01')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: The timezone provided to builtin function todatetimeoffset is invalid.)~~
+
+
+Select todatetimeoffset(CAST('1900-05-06 13:59:29.998 -8:00' AS datetime2), '+12:00')
```

Comment:
what happened to -8:00?

Reply:
That will be replaced by the new timezoneoffset as expected by sql server
```
SQL SERVER
1> Select todatetimeoffset(CAST('1900-05-06 13:59:29.998 -8:00' AS datetime2), '+12:00')
2> go
                                             
---------------------------------------------
           1900-05-06 13:59:29.9980000 +12:00
```